packages/plugin-ext/src/common/known-commands.ts

```diff
@@ -0,0 +1,329 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Range as R, Position as P, Location as L } from 'vscode-languageserver-types';
+import URI from 'vscode-uri';
+import * as theia from '@theia/plugin';
+
+// Here is a mapping of VSCode commands to monaco commands with their conversions
+export namespace KnownCommands {
+
+    /**
+     * Commands that you want to apply custom conversions to rather than pass through the automatic args converter.
+     * Would be useful in the case where theia provides some command and you need to provide custom conversions
+     */
+    // tslint:disable-next-line:no-any
+    export const transformers: { [id: string]: [string, (args: any[] | undefined) => any[] | undefined] } = {};
+    transformers['editor.action.showReferences'] = ['textEditor.commands.showReferences', createConversionFunction(
+        (uri: URI) => uri.toString(),
+        fromPositionToP,
+        toArrayConversion(fromLocationToL))];
+
+    export function mapped(id: string): boolean {
+        return !!mappings[id];
+    }
+
+    /**
+     * Commands that have a name in theia that is different from their name in vscode
+     */
+    export const mappings: { [id: string]: string } = {};
```

Comment:
You really don't need a second array here: the "transformers" array already contains the mapping. The idea I have was this: if you just pass an empty array into "createConversionFunction", it will return the identity function (no conversion).  I think we can trust the JIT to optimize this case (if you are worried about performance) So 
transformers['id']= ['mappedId', createConversionFunction([])] should do the trick.

Also, memory is not an argument, as we are talking about < 100*8 bytes (100 references)